pkg/riskanalysis/cmd.go

```diff
@@ -0,0 +1,96 @@
+package riskanalysis
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+)
+
+// Options is used to run a risk analysis to determine how severe or unusual
+// the test failures in an openshift-tests run were.
+type Options struct {
+	Out, ErrOut io.Writer
+	JUnitDir    string
+}
+
+const testFailureSummaryFilePrefix = "test-failures-summary"
+
+// Run performs the test risk analysis by reading the output files from the test run, submitting them to sippy,
+// and writing out the analysis result as a new artifact.
+func (opt *Options) Run() error {
+	fmt.Fprintf(opt.Out, "Scanning for %s files in: %s\n", testFailureSummaryFilePrefix, opt.JUnitDir)
+
+	resultFiles, err := filepath.Glob(fmt.Sprintf("%s/%s*.json", opt.JUnitDir, testFailureSummaryFilePrefix))
+	if err != nil {
+		return err
+	}
+	fmt.Fprintf(opt.Out, "Found files: %v\n", resultFiles)
+
+	prowJobRuns := []*ProwJobRun{}
+	// Read each result file into a ProwJobRun struct:
+	for _, rf := range resultFiles {
+		data, err := os.ReadFile(rf)
+		if err != nil {
+			return err
+		}
+		jobRun := &ProwJobRun{}
+		err = json.Unmarshal(data, jobRun)
+		if err != nil {
+			return errors.Wrapf(err, "error unmarshalling ProwJob json")
+		}
+		prowJobRuns = append(prowJobRuns, jobRun)
+	}
+
+	// We will often have more than one output file for this job run because openshift-tests is often
+	// invoked multiple times (pre/post upgrade). We need to merge the data together in this case.
+	var finalProwJobRun *ProwJobRun
+	for _, pjr := range prowJobRuns {
+		if finalProwJobRun == nil {
+			finalProwJobRun = pjr
+			continue
+		}
+		if pjr.ProwJob.Name != finalProwJobRun.ProwJob.Name {
+			return fmt.Errorf("mismatched job names found in %s files, %s != %s",
+				testFailureSummaryFilePrefix, finalProwJobRun.ProwJob.Name, pjr.ProwJob.Name)
+		}
+		finalProwJobRun.Tests = append(finalProwJobRun.Tests, pjr.Tests...)
+	}
+
+	// TODO: query sippy
+	url := "http://localhost:8080/api/jobs/runs/risk_analysis"
```

Comment:
Would we want this to be a config option so we could change down the road without updating the code?

Reply:
Oops, that should be the real URL.

If sippy moves we'll have to make a commit somewhere, would that be better in release repo to update config, or in origin? Release repo might be  a little better as it's probably exempt from code freeze. I could do a CLI flag for this with a default, so it's there if we ever need it.

Reply:
Having it in the test step (maybe with a default here) makes sense. If it ever changes we'll avoid having to backport the changes

Reply:
I like the idea of cli flag that can be defaulted and then you can change it in release repo when you call this as a step.  Also makes it nice so you can call it locally for testing.